tests/unit/lib/DynamicArray_unittest.cc

```diff
@@ -8,265 +8,178 @@
 using namespace shogun;
 
 template <typename T>
-class CDynamicArrayTest : public ::testing::Test
+class CDynamicArrayFixture : public ::testing::Test
 {
 protected:
-	CDynamicArrayTest()
+	CDynamicArrayFixture()
 	{
 	}
 	virtual void SetUp()
 	{
-		default_array = new CDynamicArray<T>();
-		custom_array = new CDynamicArray<T>(5);
-	}
-	virtual void TearDown()
-	{
-		SG_FREE(default_array);
-		SG_FREE(custom_array);
-	}
-	virtual ~CDynamicArrayTest()
-	{
-	}
-	CDynamicArray<T>* default_array;
-	CDynamicArray<T>* custom_array;
-};
-
-class CDynamicArrayFixture : public ::testing::Test
-{
-protected:
-	CDynamicArrayFixture()
-	{
-		b_array = SG_MALLOC(bool, 5);
-		i_array = SG_MALLOC(int32_t, 5);
+		array = SG_MALLOC(T, 5);
 		for (int32_t i = 0; i < 5; i++)
 		{
-			b_array[i] = true;
-		}
-		for (int32_t i = 0; i < 5; i++)
-		{
-			i_array[i] = i;
+			array[i] = (T)i;
 		}
-	}
-	virtual void SetUp()
-	{
-		wrapper_array_b = new CDynamicArray<bool>(this->b_array, 5);
-		wrapper_array_i = new CDynamicArray<int32_t>(this->i_array, 5);
+		wrapper_array = new CDynamicArray<T>(this->array, 5);
 	}
 	virtual void TearDown()
 	{
-		SG_FREE(wrapper_array_b);
-		SG_FREE(wrapper_array_i);
+		SG_UNREF(wrapper_array);
+		SG_FREE(array);
```

Comment:
mmm, I think we still need to free the array. Otherwise we will lost some memory https://pastebin.mozilla.org/9026080. 

And if i change it to 

	virtual void SetUp()
	{
		array = SG_MALLOC(T, 5);
		for (int32_t i = 0; i < 5; i++)
		{
			array[i] = (T)i;
		}
		wrapper_array = new CDynamicArray<T>(m_array, 5);
		SG_FREE(array);
	}

The memory check will be ok https://pastebin.mozilla.org/9026081